packages/core/admin/server/src/services/homepage.ts

```diff
@@ -0,0 +1,131 @@
+   */
+  const coreStore = strapi.db.query('strapi::core-store');
+
+  return {
+    async getRecentUpdates(): Promise<GetRecentDocuments.Response['data']> {
+      // Get all the content types the user has permissions to read
+      const readPermissions = await permissionService.findMany({
+        where: {
+          role: { users: { id: strapi.requestContext.get()?.state?.user.id } },
+          action: 'plugin::content-manager.explorer.read',
+        },
+      });
+      const allowedContentTypeNames = readPermissions
+        .map((permission) => permission.subject)
+        .filter(Boolean) as UID.ContentType[];
+
+      // Fetch the configuration for each content type in a single query
+      const rawConfigurations = await coreStore.findMany({
+        where: {
+          key: {
+            $in: allowedContentTypeNames.map(
+              (contentType) => `plugin_content_manager_configuration_content_types::${contentType}`
+            ),
+          },
+        },
+      });
+      const configurations = rawConfigurations.map((rawConfiguration) => {
+        return JSON.parse(rawConfiguration.value);
+      });
+
+      const recentDocuments = await Promise.all(
+        allowedContentTypeNames.map(async (contentTypeName) => {
+          const configuration = configurations.find((config) => config.uid === contentTypeName);
+          const contentType = strapi.contentType(contentTypeName);
+          const fields = ['documentId', 'updatedAt'];
+
+          // Add fields required to get the status if D&P is enabled
+          const hasDraftAndPublish = contentTypes.hasDraftAndPublish(contentType);
+          if (hasDraftAndPublish) {
+            fields.push('publishedAt');
+          }
+
+          // Only add the main field if it's defined
+          const { mainField } = configuration.settings;
+          if (mainField) {
+            fields.push(mainField);
+          }
+
+          // Only add locale if it's localized
+          const isLocalized = (contentType.pluginOptions?.i18n as any)?.localized;
+          if (isLocalized) {
+            fields.push('locale');
+          }
+
+          const documents = await strapi.documents(contentTypeName).findMany({
+            limit: MAX_DOCUMENTS,
+            sort: 'updatedAt:desc',
+            fields,
+          });
+
+          return documents.map((document) => {
+            /**
+             * Save the main field value before deleting it so we can use the common
+             * title key instead across all content types. Use the delete operator instead of
+             * destructuring or lodash omit for better type inference.
+             */
+            const mainFieldValue = document[mainField ?? 'documentId'];
+            delete document[mainField];
+
+            return {
+              data: {
+                ...document,
+                updatedAt: new Date(document.updatedAt),
+                title: mainFieldValue,
+              },
```

Comment:
Could the delete just be avoided with something like this? 

```suggestion
            const { mainField, ...restDcoument } = document;

            return {
              data: {
                ...restDocument,
                updatedAt: new Date(document.updatedAt),
                title: mainField ?? 'documentId',
              },
```

Reply:
lemme try again but I think the types weren't happy when I tried that

Reply:
Actually sorry this does break type safety. The reason is that mainField is dynamic, so when I destructure `[mainField]: mainFieldValue`, typescript thinks that because mainField is dynamic then nothing is safe anymore. Not 100% sure why I don't have the issue with `delete` though 🤷

Reply:
No worries, I'm cool with what you have too just thought I'd suggest this as an option, but if TS isn't happy let's stick with what you have

Reply:
Yeah I'm sad it doesn't work as well because it's very clean